pkg/resource/azurepathfixjob.go

```diff
@@ -99,7 +99,15 @@ func (gapfj *generatorAzurePathFixJob) expected() (runtime.Object, error) {
 		{Name: "AZURE_CONTAINER_NAME", Value: azureStorage.Container},
 		{Name: "AZURE_CLIENT_ID", Value: azureCfg.ClientID},
 		{Name: "AZURE_TENANT_ID", Value: azureCfg.TenantID},
-		{Name: "AZURE_CLIENT_SECRET", Value: azureCfg.ClientSecret},
+		{Name: "AZURE_CLIENT_SECRET", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				Optional: &optional,
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: defaults.CloudCredentialsName,
```

Comment:
What will happen if Azure is configured via `REGISTRY_STORAGE_AZURE_ACCOUNTKEY` in the user configuration secret?

Reply:
just realized you asked about the user configuration secret - that won't actually work unless the user manually updates the non-user secret, we have a separate bug for that https://issues.redhat.com/browse/OCPBUGS-32703, but this behavior is very old so it's a low priority fix.